rotkehlchen/chain/ethereum/modules/liquity/decoder.py

```diff
@@ -206,12 +197,42 @@ def _decode_lqty_staking_deposits(
 
     # -- DecoderInterface methods
 
-    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
-        return {
-            ACTIVE_POOL: (self._decode_trove_operations,),
-            STABILITY_POOL: (self._decode_stability_pool_event,),
-            LIQUITY_STAKING: (self._decode_lqty_staking_deposits,),
-        }
+    def _handle_post_decoding(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+    ) -> list[HistoryBaseEntry]:
+        """
+        Handles post decoding for liuity. We handle the entire decoding in a post-decoding rule
+        because "normal" decoding will not work well if a DS Proxy is used.
+
+        In case of usage of a DS Proxy transfers `proxy` <-> `liquity contracts` and
+        transfers `proxy` <-> `user address` are in a bad order, so we have to have all erc20 / eth
+        transfers decoded before applying liquity-specific logic.
+
+        Can't use action items here because action items require to know all properties of an
+        event that we expect to appear in the future, and we unfortunately can't know them.
+        """
+        # TODO: when post decoding rules are refactored, make sure that these rules run only
+        # if liquity contracts were touched.
+        for tx_log in all_logs:
```

Comment:
Isn't this expensive? I'm facing a similar roadblock in curve zap decoder PR...
Looping through all transaction logs and for each iteration, there's a nested loop that goes through every decoded event.